### DISCUSSION TITLE
Validate anchor links?

### DISCUSSION THREAD

**Discussion by johnthagen** [squidfunk/mkdocs-material#2699] [Ideas]
An upstream issue to MkDocs to validate the `#anchor` portion of links was closed citing that a third party plugin should do this: https://github.com/mkdocs/mkdocs/issues/658

Is this something within scope for Material for MkDocs? Is there some way to solve the general problem of:

```md
# Title

See the following [Section](#title-3).

# Title 2
```

where `#title-3` is an accidental typo, but everything builds without warnings? To me this seems like a pretty big footgun towards keeping documentation accurate. For comparison, Sphinx can detect invalid links and throw an error.

Is there any solution to catch these types of errors during the build?

**Comment by squidfunk** [comment]
In general, Material for MkDocs is a theme and only started to provide plugins as part of Insiders (i.e. the [built-in tags plugin](https://squidfunk.github.io/mkdocs-material/setup/setting-up-tags/#built-in-tags)). Those plugins are tailored to Material for MkDocs and may not be useful for other themes without further ado. Since anchor validation is something entirely non-theme related, I'd say it should be solved as part of a dedicated third-party plugin, so it's out of scope for this project.

It's always a good idea to check the [MkDocs plugin wiki](https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins), maybe there's already a plugin providing this functionality.

**Comment by johnthagen** [comment]
Anchor validation has been released into [`mkdocs-htmlproofer-plugin` version 0.5.0](https://github.com/manuzhang/mkdocs-htmlproofer-plugin/releases/tag/v0.5.0).

**Comment by mschoettle** [comment]
Mkdocs 1.6 can validate anchors now. See the `anchors` validation option: https://www.mkdocs.org/user-guide/configuration/#validation
